### PR TITLE
docs: Add Persistent Workload using Host Volumes guide 

### PR DESCRIPTION
**Pull request by angrycub**
Based on the existing portworx guide

depends on #6211

### CODE REVIEW

File: website/source/guides/stateful-workloads/host-volumes.md [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+### Step 2: Configure the `mysql` Host Volume on the Client
+
+Edit the Nomad configuration on this Nomad client to create the Host Volume.
+
+Add the following to the `client` stanza of your Nomad configuration:
+
+```hcl
+  host_volume "mysql" {
+    path="/data/mysql"
+    read_only="false"
```
Comment (endocrimes) [inline website/source/guides/stateful-workloads/host-volumes.md]: spacing around `=` and drop the quotes on `false` for consistency/clarity?

File: website/source/guides/stateful-workloads/host-volumes.md [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+  }
+```
+
+Save this change, and then restart the Nomad service on this client to make the
+Host Volume active.
```
Comment (endocrimes) [inline website/source/guides/stateful-workloads/host-volumes.md]: Potentially include a "You can verify this by running `nomad node status -short <node-id>`"?

File: website/source/guides/stateful-workloads/host-volumes.md [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+
+### Step 3: Create the `mysql.nomad` Job File
+
+We are now ready to deploy a MySQL database that can use Nomad Host Volumes for
+storage. Create a file called `mysql.nomad` and provide it the following
+contents:
+
+```hcl
```
Comment (endocrimes) [inline website/source/guides/stateful-workloads/host-volumes.md]: Can we run the hcl through hclfmt`?

File: website/source/guides/stateful-workloads/host-volumes.md [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+==> Evaluation "6b784149" finished with status "complete"
+```
+
+Verify no jobs are running in the cluster:
+
+```
+$ nomad status
+No running jobs
+```
+You can optionally stop the nomad service on whichever node you are on and move
```
Comment (endocrimes) [inline website/source/guides/stateful-workloads/host-volumes.md]: this wouldn't necessarily work with host volumes unless they had external data replication

File: website/source/guides/stateful-workloads/stateful-workloads.html.md [hunk 5]
```diff
@@ -9,6 +9,14 @@ description: |-
 
 # Stateful Workloads
 
+Nomad allows a user to mount persistent data from local or remote storage volumes
+into task environments in a couple of ways—host volume mounts or Docker Volume
```
Comment (endocrimes) [inline website/source/guides/stateful-workloads/stateful-workloads.html.md]:
```suggestion
into task environments in a couple of ways — host volume mounts or Docker Volume
```

File: website/source/guides/stateful-workloads/host-volumes.md [hunk 6]
```diff
@@ -0,0 +1,338 @@
+---
+layout: "guides"
+page_title: "Stateful Workloads with Nomad Host Volumes"
+sidebar_current: "guides-stateful-workloads"
+description: |-
+  There are multiple approaches to deploying stateful applications in Nomad.
+  This guide uses Nomad Host Volumes deploy a MySQL database.
```
Comment (preetapan) [inline website/source/guides/stateful-workloads/host-volumes.md]: This guide uses Nomad Host Volumes to deploy a MySQL database,

File: website/source/layouts/guides.erb [hunk 7]
```diff
@@ -207,6 +207,9 @@
       <li<%= sidebar_current("guides-stateful-workloads") %>>
         <a href="/guides/stateful-workloads/stateful-workloads.html">Stateful Workloads</a>
         <ul class="nav">
+          <li<%= sidebar_current("guides-portworx") %>>
```
Comment (preetapan) [inline website/source/layouts/guides.erb]: sidebar name should be `guides-stateful-workloads`

File: website/source/guides/stateful-workloads/host-volumes.md (hunk 4 above)
Comment (Omar-Khawaja) [reply website/source/guides/stateful-workloads/host-volumes.md]: I've resolved the other issues, but I'm not clear on this particular issue. Can this line simply be removed for accuracy or is there another issue I'm missing?

File: website/source/guides/stateful-workloads/stateful-workloads.html.md [hunk 8]
```diff
@@ -9,6 +9,14 @@ description: |-
 
 # Stateful Workloads
 
+Nomad allows a user to mount persistent data from local or remote storage volumes
+into task environments in a couple of ways — host volume mounts or Docker Volume
+drivers.
+
```
Comment (rkettelerij) [inline website/source/guides/stateful-workloads/stateful-workloads.html.md]: Can you describe the advantage of using host volumes as opposed to using Docker volumes (with the default volume driver)? The example deploys a MySQL server running in Docker. It's not immediately clear what benefit host volumes bring in this case.

Comment (angrycub) [reply website/source/guides/stateful-workloads/stateful-workloads.html.md]: Thanks for the suggestion; I appreciate it! The primary value proposition is that host volumes are a more taskdriver-agnostic approach to volume handling.  Nomad is also fully aware of them for the purposes of dynamic, implicit constraints around their availability.  I will add some prose at the top to see if I can highlight that. I am hoping to highlight more as we grow the documentation around host volumes as well.

File: website/source/guides/stateful-workloads/host-volumes.md (hunk 4 above)
Comment (angrycub) [reply website/source/guides/stateful-workloads/host-volumes.md]: I missed this line in the sweep. 🤦‍♂ Deleted it; added new copy to explain how HA might be supported by a mount over the directory in question.

File: website/source/guides/stateful-workloads/stateful-workloads.html.md (hunk 8 above)
Comment (endocrimes) [reply website/source/guides/stateful-workloads/stateful-workloads.html.md]: We could also add a note on the fact they have an ACL system

**Pull request merged**